stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/core.py

```diff
@@ -355,12 +360,14 @@ async def create_item(self, item: stac_types.Item, **kwargs) -> stac_types.Item:
     async def update_item(self, item: stac_types.Item, **kwargs) -> stac_types.Item:
         """Update item."""
         base_url = str(kwargs["request"].base_url)
+        collection_id = item["collection"]
+
         now = datetime_type.now(timezone.utc).isoformat().replace("+00:00", "Z")
         item["properties"]["updated"] = str(now)
 
-        await self.database.check_collection_exists(collection_id=item["collection"])
+        await self.database.check_collection_exists(collection_id)
         # todo: index instead of delete and create
-        await self.delete_item(item_id=item["id"], collection_id=item["collection"])
+        await self.delete_item(item_id=item["id"], collection_id=collection_id)
```

Comment:
Is there any race condition here? E.g. the collection is removed between 368 and 370?

Reply:
Possibly, in that case, it may return a 500 instead of a 404. This issue is already open:

- #75 

to improve the update behavior, since update should be updating the existing ES document by ID instead of delete and create.